### PR TITLE
feat: add support for job_options argument

### PR DESCRIPTION
**Pull request by bautrey37**
In Que [v1.2](https://github.com/que-rb/que/pull/336), it was deprecated to pass top level keyword args in the
`enqueue` method. This is to prepare for Que v2. This causes warnings
such as:

```
Passing job options like (tags) to `JobClass.enqueue` as top level keyword args has been deprecated and will be removed in version 2.0. Please wrap job options in an explicit `job_options` keyword arg instead.
```

The keyword arguments are: `queue`, `priority`, `run_at`, `tags`, and
`job_class`. These must be wrapped around the `job_options` keyword.

### CODE REVIEW

File: instrumentation/que/lib/opentelemetry/instrumentation/que/patches/que_job.rb [hunk 1]
```diff
@@ -37,12 +37,14 @@ def enqueue(*args, tags: nil, **arg_opts)
...
                 # much safer to use than modifying the payload.
+                tags = job_options[:tags]
+                job_class = job_options[:job_class]
                 if otel_config[:propagation_style] != :none
                   tags ||= []
                   OpenTelemetry.propagation.inject(tags, setter: TagSetter)
                 end
 
-                job = super(*args, tags: tags, **arg_opts)
+                job = super(*args, job_options: {tags: tags, job_class: job_class}, **arg_opts)
```
Comment (indrekj) [inline instrumentation/que/lib/opentelemetry/instrumentation/que/patches/que_job.rb:47]: Everything in `job_options` has to passed to super as well. Imagine `job_options` had a key `foo`. That would not be discarded.

File: instrumentation/que/lib/opentelemetry/instrumentation/que/instrumentation.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@ module Que
       # The Instrumentation class contains logic to detect and install the Que
       # instrumentation
       class Instrumentation < OpenTelemetry::Instrumentation::Base
-        MINIMUM_VERSION = Gem::Version.new('1.0.0.beta4')
+        MINIMUM_VERSION = Gem::Version.new('1.2.0')
```
[the remark this answers is not in the document]
Comment (indrekj) [reply instrumentation/que/lib/opentelemetry/instrumentation/que/instrumentation.rb:13]: Looks like 1.2.0 is indeed correct. They seemed to have changed the enqueue signature in that versin.

File: instrumentation/que/lib/opentelemetry/instrumentation/que/patches/que_job.rb (hunk 1 above)
Comment (indrekj) [reply instrumentation/que/lib/opentelemetry/instrumentation/que/patches/que_job.rb:47]: E.g.

```
job_options = job_options.merge(tags: tags)
super(*args, job_options: job_options, **arg_opts)
```
or smth similar.

Also, I don't think you need to extract `job_class` in that case either

Comment (bautrey37) [reply instrumentation/que/lib/opentelemetry/instrumentation/que/patches/que_job.rb]: True. This code works. [changes[(https://github.com/open-telemetry/opentelemetry-ruby/compare/a74bf38cd592e811540fc0db05714157b54678bb..6f89f73e2da1b884c75d184be9d8268d97cffac1)

**Comment by bautrey37** [comment]
This is my first PR here. I believe this should be a minor version bump since the underlying gem `que` is being upgraded. Upgrading `que` to the latest v1, would require additional changes such as migrating the que database, so v1.2 is the acceptable minimum version for now.

**Comment by plantfansam** [comment]
Thanks so much for the contributions @bautrey37 and @indrekj! We're definitely on the right track here! It looks like some of the specs are failing, though. The [test output](https://github.com/open-telemetry/opentelemetry-ruby/runs/6408338574?check_suite_focus=true) can be a bit hard to read, but if you search for `7 failures` you should be able to track them down. Mind taking a look? Happy to assist with debugging 😄.

**Comment by bautrey37** [comment]
@plantfansam Can you assist me? I updated the branch, and now the build checks that were there, are no longer there. What happened to them?

**Comment by indrekj** [comment]
@bautrey37 I think your tests are failing because you haven't updated the Appraisals file

**Comment by plantfansam** [comment]
Hello, and thank you for your contribution!

We recently split Ruby instrumentation out into the [`opentelemetry-ruby-contrib` repo](https://github.com/open-telemetry/opentelemetry-ruby-contrib).

This PR is related to instrumentation, so we'll need you to re-open it against `opentelemetry-ruby-contrib`. Sorry for the inconvenience! 

To do that, you can:

1. [Create a fork of `opentelemetry-ruby-contrib`](https://github.com/open-telemetry/opentelemetry-ruby-contrib/fork) and copy the git url
1. In your `opentelemetry-ruby` repo, run `git remote add tmp-contrib <your-fork-git-url>`
1. `git push tmp-contrib your-branch-name`
1. Open a new PR in contrib (feel free to just copy/paste your original PR description there)
1. Close your open PR in this repo with a comment that links to your new PR in contrib
1. Delete your `tmp-contrib` remote from `opentelemetry-ruby` (`git remote rm tmp-contrib`)
1. `git clone` your `opentelemetry-ruby-contrib` fork, check out your branch, and make all changes in that repo from now on!

Sorry again for the inconvenience, and thank you for contributing!

**Comment by bautrey37** [comment]
@plantfansam Thank you for the help. I've opened up the PR on the contrib repo. Can you take a look? Thanks.

**Pull request closed without merging**